### PR TITLE
AIP-69: Adding Edge Provider Package for Edge Executor / Worker

### PR DESCRIPTION
**Pull request by jscheffl**
After a PoC in PR #40224 and the ongoing implementation which in total as "Mothership" was tracked in PR #40900 here is the AIP-69 MVP Implementation PR.

UPDATE: After completion of the discussion the "Remote Executor" was now renamed to "Edge Executor" in the PR.

This PR contains all files needed to make the provider package for AIP-69 a MVP == Minimum Viable Product. Besides (only) provider files the core adjustments are contained which are needed to make CI happy.

As greed in the AIP-69 vote the Edge provider will be marked as pre-release and as experimental. Also docs are pointing to this.

As it is a MVP, it can be started to be experienced and functions are limited to the bare minimum. But as something that is working, further functions can be added. Status of implementation is tracked in `docs/apache-airflow-providers-edge/edge_executor.rst` where also limitations are listed.

As this is a split for the provider-only files as PR from the overall PR #40900, there are two PRs completing the set which are tracked separately:
- PR for changes in Core, independent of the provider package: #41730
- PR for development convenience in Breeze: #41731

If you want to have something "testable" being running, please use descriptions and code from PR #40900

FYI @clellmann @wolfdn @AutomationDev85 @OliverWannenwetsch

### CODE REVIEW

File: airflow/providers/edge/provider.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+# under the License.
+
+# yaml-language-server: $schema=../../provider.yaml.schema.json
+---
+package-name: apache-airflow-providers-edge
+name: Edge Executor
+description: |
+  Handle edge workers on remote sites via HTTP(s) connection and orchestrates work over distributed sites
+
+state: ready
```
Comment (potiuk) [inline airflow/providers/edge/provider.yaml]: This should be "not-ready".

**Comment by potiuk** [comment]
Maybe we can further split that one? It's a **huge** one

**Comment by jscheffl** [comment]
> Maybe we can further split that one? It's a **huge** one

Yeah but actually... all individual pieces are pretty "simple". Please have a look to the 6 split PRs I cut from this. Needed to cumulate a bit to make tests working with dependencies between the slices though... each PR check only the last commit or in order once merged the LoC is reducing if we rebase incrementally.

**Comment by jscheffl** [comment]
All pieces of the PR were merged individually, this is now done implicitly!

**Pull request closed without merging**